app/code/Magento/Directory/Model/Config/Source/Country.php

```diff
@@ -44,17 +44,17 @@ public function __construct(\Magento\Directory\Model\ResourceModel\Country\Colle
      */
     public function toOptionArray($isMultiselect = false, $foregroundCountries = '')
     {
-        if (!$this->_options) {
-            $this->_options = $this->_countryCollection->loadData()->setForegroundCountries(
+        $options = $this->_options;
+        if (!$options) {
+            $options = $this->_countryCollection->loadData()->setForegroundCountries(
```

Comment:
Hello @manavluhar, thanks for submitting your PR.

I think some logic is missing here: in the deleted code, we had an assignment to `$this->_options` that has been removed in this PR.

The purpose of that assignment was that of somewhat persisting the value in the object state, something that the new logic doesn't do anymore.

Thus, I would not let this PR get in.

Let me know if it makes sense.

Cheers, 
Alessandro

Reply:
Got your point here @aleron75
Thank you for your review 😄